File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Controllers/CollectionViewControllers/CollectionViewCells/BtImageCollectionViewCell.cs


using System;

using Foundation;
using UIKit;

namespace bugTrapKit
{
	public partial class BtImageCollectionViewCell : UICollectionViewCell
	{
		public BtImageCollectionViewCell (IntPtr handle) : base(handle)
		{
		}

		public bool IsSelected { get; set; }

		public void SetData (UIImage image)
		{
			ImageView.Image = image;
		}

		public void SetSelected (bool isSelected)
		{
			IsSelected = isSelected;
			SelectedOverlay.Hidden = !isSelected;
		}
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/Controllers/CollectionViewControllers/CollectionViewCells/BtImageCollectionViewCell.swift
//
//  BtImageCollectionViewCell.swift
//  bugTrap
//
//  Created by Colby L Williams on 10/28/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import UIKit

class BtImageCollectionViewCell: UICollectionViewCell {
	
	@IBOutlet var ImageView: UIImageView!
	@IBOutlet var SelectButton: UIButton!
	@IBOutlet var SelectedOverlay: UIView!
	
	var lIsSelected = false
	
	func setData(image: UIImage?)
	{
		ImageView.image = image
	}
	
	func setIsSelected (isSelected: Bool)
	{
		// selected = isSelected
		self.lIsSelected = isSelected
		SelectedOverlay.hidden = !isSelected
	}
}
